basis/http/client/client.factor

```diff
@@ -111,15 +142,73 @@ SYMBOL: redirects
     "transfer-encoding" header "chunked" =
     [ read-chunked ] [ each-block ] if ; inline
 
+: request-socket-endpoints ( request -- physical logical )
+    [ proxy-url>> ] [ url>> ] bi [ or ] keep ;
+
 : <request-socket> ( -- stream )
-    request get url>> url-addr ascii <client> drop
+    request get request-socket-endpoints [ url-addr ] bi@
+    remote-address set ascii <client> local-address set
     1 minutes over set-timeout ;
 
+: https-tunnel? ( request -- ? )
+    [ proxy-url>> ] [ url>> protocol>> "https" = ] bi and ;
```

Comment:
It uses username and password. Later versions might also use the protocol..
So using urls is nice?
Le 8 mars 2016 16:57, "John Benediktsson" <notifications@github.com> a
écrit :

> In basis/http/client/client.factor
> <https://github.com/factor/factor/pull/1531#discussion_r55379544>:
>
> >      1 minutes over set-timeout ;
> >
> > +: https-tunnel? ( request -- ? )
> > +    [ proxy-url>> ] [ url>> protocol>> "https" = ] bi and ;
>
> Should proxy-url be a string? Since we only want the proxy to use host
> and port?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/factor/factor/pull/1531/files#r55379544>.
>